bootstrap/bootstrap/src/bootstrap.rs

```diff
@@ -57,26 +59,15 @@ fn boostrap_zos(cfg: &config::Config) -> Result<()> {
 
     // write down boot info for other system components (like upgraded)
     flist.write(FLIST_INFO_FILE)?;
-    std::fs::write(FLIST_NAME_FILE, format!("{}/{}", FLIST_REPO, flist.name))?;
+    std::fs::write(FLIST_NAME_FILE, format!("{}/{}", ZOS_REPO, flist.name))?;
 
     install_package(&flist)
 }
 
-/// bootstrap stage install and starts all zos daemons
-pub fn bootstrap(cfg: &config::Config) -> Result<()> {
-    debug!("runmode: {:?}", cfg.runmode);
-    let result = WorkDir::run(WORKDIR, || -> Result<()> {
-        boostrap_zos(cfg)?;
-        Ok(())
-    })?;
-
-    result
-}
-
 /// update stage make sure we are running latest
 /// version of bootstrap
 pub fn update(cfg: &config::Config) -> Result<()> {
-    let result = WorkDir::run(WORKDIR, || -> Result<()> {
+    let result = WorkDir::run(WORKDIR, || {
```

Comment:
I don't fully understand how errors are handled in Rust, but inferring the return type for closure will allow to return anyhow propagated errors coming from `update_bootstrap()?`
Is it necessary or will this happen by default?

Reply:
If you check the `run` function signature you will see that it takes a closure (FnOnce) that return Result<T>. Where T is a generic type.

The generic is usually inferred from the code, if the compiler fail to infer it, you will have to define it yourself (like bfeore)

so if I passed a clousre `|| { 10  }` the compiler will assume the return type is a `usize` and then the return type is Result<usize>. 

The code has enough clause so the compiler knows what I return, hence I don't have to define it as before. which is `Result<(), anyhow::Error>`